extensions/amp-script/0.1/test/unit/test-amp-script.js

```diff
@@ -230,46 +230,68 @@ describes.fakeWin('AmpScript', {amp: {runtimeOn: false}}, env => {
   });
 });
 
-describes.fakeWin('AmpScriptService', {amp: {runtimeOn: false}}, env => {
-  let crypto;
-  let service;
+describes.repeated(
+  '',
+  {
+    'single ampdoc': {ampdoc: 'single'},
+    'shadow ampdoc': {ampdoc: 'shadow'},
+  },
+  (name, variant) => {
+    describes.fakeWin(
+      'AmpScriptService',
+      {
+        amp: {
+          runtimeOn: false,
+          ampdoc: variant.ampdoc,
+        },
+      },
+      env => {
+        let crypto;
+        let service;
 
-  beforeEach(() => {
-    crypto = {sha384Base64: env.sandbox.stub()};
-    env.sandbox.stub(Services, 'cryptoFor').returns(crypto);
-  });
+        beforeEach(() => {
+          crypto = {sha384Base64: env.sandbox.stub()};
+          env.sandbox.stub(Services, 'cryptoFor').returns(crypto);
+        });
 
-  function createMetaTag(name, content) {
-    const meta = document.createElement('meta');
-    meta.setAttribute('name', name);
-    meta.setAttribute('content', content);
-    env.ampdoc.getHeadNode().appendChild(meta);
-  }
+        function createMetaHash(name, content) {
+          if (variant.ampdoc === 'shadow') {
+            env.ampdoc.setMetaByName(name, content);
```

Comment:
nit, wouldn't this work for single doc also?

Reply:
Thanks for the review. AmpDocSingle does not implement this method since the meta tags are expected to exist in <head>. `.setMetaByName()` is shadow-only for now.

Reply:
Got it, thanks. I'll defer to the original discussion but it might be worth considering in the future to unify the API